examples/cu_consolemon/src/sysinfo.rs

```diff
@@ -0,0 +1,220 @@
+            "cpu" => Ok(PfetchInfo::Cpu),
+            "memory" => Ok(PfetchInfo::Memory),
+            "shell" => Ok(PfetchInfo::Shell),
+            "editor" => Ok(PfetchInfo::Editor),
+            "wm" => Ok(PfetchInfo::Wm),
+            "de" => Ok(PfetchInfo::De),
+            "palette" => Ok(PfetchInfo::Palette),
+            unknown_info => Err(format!("Unknown pfetch info: {unknown_info}")),
+        }
+    }
+}
+
+// Struct for the various readouts
+pub struct Readouts {
+    pub general_readout: GeneralReadout,
+    pub package_readout: PackageReadout,
+    pub memory_readout: MemoryReadout,
+    pub kernel_readout: KernelReadout,
+}
+
+pub fn get_info(info: &PfetchInfo, readouts: &Readouts) -> Option<String> {
+    match info {
+        PfetchInfo::Ascii => None,
+        PfetchInfo::Title => pfetch::user_at_hostname(
+            &readouts.general_readout,
+            &dotenvy::var("USER").ok(),
+            &dotenvy::var("HOSTNAME").ok(),
+        ),
+        PfetchInfo::Os => pfetch::os(&readouts.general_readout),
+        PfetchInfo::Host => pfetch::host(&readouts.general_readout),
+        PfetchInfo::Kernel => pfetch::kernel(&readouts.kernel_readout),
+        PfetchInfo::Uptime => pfetch::uptime(&readouts.general_readout),
+        PfetchInfo::Cpu => pfetch::cpu(&readouts.general_readout),
+        PfetchInfo::Memory => pfetch::memory(&readouts.memory_readout),
+        PfetchInfo::Shell => pfetch::shell(&readouts.general_readout),
+        PfetchInfo::Editor => Some(env::var("EDITOR").unwrap_or_else(|_| "nvim".to_string())),
+        PfetchInfo::Wm => pfetch::wm(&readouts.general_readout),
+        PfetchInfo::De => pfetch::de(&readouts.general_readout),
+        PfetchInfo::Palette => Some("Color Palette".to_string()), // Simplified palette display
+        PfetchInfo::BlankLine => Some("".to_string()),
+    }
+}
+
+// Function to render the gathered info along with ASCII logo
+pub fn pfetch_info() -> String {
+    let readouts = Readouts {
+        general_readout: GeneralReadout::new(),
+        package_readout: PackageReadout::new(),
+        memory_readout: MemoryReadout::new(),
+        kernel_readout: KernelReadout::new(),
+    };
+
+    let os = pfetch::os(&GeneralReadout::new()).unwrap_or_default();
+
+    let all_infos = [
+        PfetchInfo::Os,
+        PfetchInfo::Host,
+        PfetchInfo::Kernel,
+        PfetchInfo::Uptime,
+        PfetchInfo::Cpu,
+        PfetchInfo::Memory,
+        PfetchInfo::Shell,
+        PfetchInfo::Editor,
+        PfetchInfo::Wm,
+        PfetchInfo::De,
+    ];
+
+    let logo = pfetch::logo(&os);
+    let gathered_pfetch_info: Vec<(Color, String, String)> = all_infos
+        .iter()
+        .filter_map(|info| match info {
+            PfetchInfo::Os => Some((logo.primary_color, info.to_string(), os.clone())),
+            _ => get_info(&info, &readouts).map(|info_str| match info {
+                PfetchInfo::Title => (logo.secondary_color, info_str, "".to_string()),
```

Comment:
MEGA nit: You can use `.into()` instead of `.to_string()` in a lot of places. Definitely a preference thing. But I like it just because it keep things a little more concise.